docs/design/enviroment_management.md

```diff
@@ -0,0 +1,156 @@
+
+Following the writing format standard above, we can achieve:
+
+- Clearly define the modules' inputs in an environment;
+- Support customized configurations for some projects;
+- Use a unify function to combine schema configurations and environment configurations to form the entire inputs for all modules. 
+
+### Commands
+
+Kusion provides the management of environment by sub command `kusion env`, the designed command is shown as below.
+
+#### kusion env init <name>
+
+Initialize an environment with specified name. An environment configuration file `<name>.yaml` in `$KUSION_HOME/.env` is created.
+
+#### kusion env rm <name>
+
+Remove an environment.
+
+#### kusion env ls
+
+List all the initialized environments.
+
+#### kusion env edit <name>
+
+Opens an editor to modify the specified environment configuration file.
+
+#### kusion env set <name> [-f|--file \<file>] [\<key> \<value>]
+
+Set the entire or the specified item of configuration of the specified environment. If setting the entire configuration, support from a file, while the specified key-value is forbidden. If setting the specified item, the item is a string which is the yaml object keys combined by dot, and the following is the example.
+
+```sh
+# set the field provider of module Database default input, environment name is dev
+kusion env set dev module.Database.default.provider aws
+
+# set the field size of module Database override fooBar input, environment name is dev
+kusion env set dev module.Database.fooBar.size 50
+```
+
+#### kusion env get <name> [\<key>]
+
+Get the entire of a specified item (if the key is specified) of the specified configuration.
+
+#### kusion env validate <name>
+
+Judge the environment configuration is valid or not.
+
+## Q&A
+
+**Q1**: Why there is a block named `projectSelector`, can it be written under the block `module_<ModuleName>`, like what is shown below?
+```yaml
+module_Database:
+  default:
+    provider: aws
+    size: 20
+    instanceClass: db.t3.micro
+    securityIPs:
+    - 10.0.0.0/18
+  fooBar:
+    size: 50
+    instanceClass: db.t3.small
+    projectSelector:
+    - foo
+    - bar
+  baz:
+    instanceClass: db.t3.large
+    projectSelector:
+    - baz  
+```
+**A1**: The format shown above can work, but there exists some drawbacks:
+
+- projectSelector becomes a keyword, it asks for the module input filed name must not be projectSelector;
+- projectSelector is at the same layer of the module input, which brings possible misunderstanding;
+- Not obey 3NF of [Database normalization](https://en.wikipedia.org/wiki/Database_normalization), there are non-key columns (module input fields) depends on other non-key columns (projectSelector), which is not a good design for the possible database storage backend in the future.
```

Comment:
I didn't understand the relationship between the `projectSelector` and 3NF, can you provide some examples?